avro-data/src/test/java/io/confluent/connect/avro/AvroDataTest.java

```diff
@@ -736,8 +735,8 @@ public void testFromConnectOptionalPrimitiveWithMetadata() {
   @Test
   public void testFromConnectRecordWithMetadata() {
     Schema schema = SchemaBuilder.struct()
-        .name("io.confluent.test.TestSchema").version(12).doc("doc")
-        .field("int32", Schema.INT32_SCHEMA)
```

Comment:
The Avro spec [https://avro.apache.org/docs/current/spec.html#schema_primitive](url) states
>  Primitive types have no specified attributes.

So only complex type schema will have `connect.record.doc`. 
Primitive type can only have `connect.field.doc` because the current implementation can't distinguish a primitive schema from being a field level schema or is itself a record level schema.